kafka/consumer_test.go

```diff
@@ -162,7 +163,11 @@ func TestConsumerInstrumentation(t *testing.T) {
 		MaxPollRecords: 1, // Consume a single record for this test.
 		Processor: apmqueue.ProcessorFunc(func(_ context.Context, r apmqueue.Record) error {
 			defer close(processed)
-			assert.Equal(t, event, r)
+			assert.Equal(t, event.Topic, r.Topic)
+			assert.Equal(t, event.Value, r.Value)
+			// We cannot know which partition the record will be polled from,
+			// but we can assert it's in the range of number of partitions.
+			assert.True(t, event.Partition < partitions-1)
```

Comment:
Would that ever not be the case? I don't follow what's being asserted here.

Reply:
The assertion is that we're setting an ID for the field `Partition` that is in the [0,n) range.
That cannot be greater than the number of partitions - 1.

Reply:
Oh maybe you're referring to the fact that `Partition` will always exist because of its zero-value?

Reply:
I think it's good to assert there's something there and is populated, that'll always be the case. A possibility for a somewhat more thorough test would be to produce enough records that they're produced to all partitions and have a test that ensures that records are received from all the partitions.

I think that'd be a good test to add, but I'll leave it up to you if you want to do it on this PR or as a follow up.